apps/music/js/ui/views/mode_manager.js

```diff
@@ -161,7 +161,14 @@ var ModeManager = {
                        'search-from-tiles-mode', 'search-from-list-mode',
                        'picker-mode'];
```

Comment:
I checked the video in bugzilla and found this could be fixed by modifying the css only without changing any JavaScript code, because it seems that the tabs for switching the modes should just appear in the proper timings, such as `tiles-mode`, `list-mode` and `sublist-mode`, and by default the tabs should be invisible and only display when some modes need it.

So I guess the patch will looks like:
https://github.com/mozilla-b2g/gaia/blob/master/apps/music/style/music.css#L914-L919
Add `display: none` to the tabs default styles and
https://github.com/mozilla-b2g/gaia/blob/master/apps/music/style/music.css#L219-L223
Set `display: inherit` with `body.tiles-mode`, `body.list-mode` and `body.sublist-mode`.

Does this make sense to you? @mancas